types.go

```diff
@@ -6,6 +6,49 @@ import (
 	"math/bits"
```

Comment:
if this file is meant to hold types relating to the fields of `Pokemon`, maybe this can be named `pokemon_field_types` or something more specific than just `types`

Reply:
Not necessarily. It's basically just types that don't have enough code to warrant having their own file.